### DISCUSSION TITLE
[Application Development] Data Indexer: What New Fields Required for Ecotone/Dencun?

### DISCUSSION THREAD

**Discussion by MSilb7** [ethereum-optimism/developers#245] [Application Development] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Today, data indexers need to pull in OP-specific fields (that don't exist in L1) in order to enable accurate DA gas & fees calculations (i.e. l1Fee, l1GasUsed, l1GasPrice).

In Ecotone, with 4844, the fee formula also needs to account for blobgas price. The requirements for data analysis may be more detailed than other consumers, since we (ideally) want to break out DA fees by L1 Commitment costs vs Blob Gas (i.e. blobgas used, blobgas price).

What fields should data providers index (and where do they find them?) on L2 and L1 for Ecotone/Dencun?

Documentation checked:
- https://docs.optimism.io/builders/node-operators/network-upgrades/ecotone
- https://docs.optimism.io/builders/notices/ecotone-fee-changes

**Comment by trianglesphere** [comment, marked as answer]
Specs Links
- https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/exec-engine.md#l1-cost-fees-l1-fee-vault
- https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/deposits.md#l1-attributes---ecotone
